packages/solarwinds-apm/src/config.ts

```diff
@@ -72,9 +72,24 @@ const serviceKey = z
     }
   })
 
-const trustedpath = z.string().transform((p, ctx) => {
+const collector = z.string().transform((c, ctx) => {
+  if (!/^https?:/.test(c)) {
+    c = `https://${c}`
+  }
   try {
-    return fs.readFile(p, "utf-8")
+    return new URL(c)
+  } catch (err) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: (err as Error).message,
+    })
+    return z.NEVER
+  }
+})
```

Comment:
Parse the collector directly to a URL, which will catch some errors early if it's set to a nonsensical value.